packages/salesforcedx-utils-vscode/test/jest/context/workspaceContextUtil.test.ts

```diff
@@ -150,5 +155,22 @@ describe('WorkspaceContext', () => {
 
       expect(connectionMock.create).toHaveBeenCalledTimes(1);
     });
+
+    it('should not throw error if there is a username set', async () => {
+      const connection = await workspaceContextUtil.getConnection();
+      expect(() => connection).not.toThrow();
+    });
+
+    it('should throw error if there is no username set', async () => {
+      getUsernameStub.mockReturnValue(undefined);
+
+      await mockWatcher.onDidChange.mock.calls[0][0]();
+
+      try {
+        await workspaceContextUtil.getConnection();
+      } catch (error) {
+        expect(error).toEqual(new Error(nls.localize('error_no_default_username')));
+      }
```

Comment:
You can use the following syntax here and avoid the issue where if the getConnection call doesn't throw the test still passes
```
      expect(() => {
        await workspaceContextUtil.getConnection();
      }).toThrowError(nls.localize('error_no_default_username'));
```

Reply:
mm that doesn't seem to be working for me🥲

Reply:
example pushed in commit